src/services/utilities.ts

```diff
@@ -2385,7 +2385,11 @@ namespace ts {
                 if (text) parts.push(linkTextPart(text));
             }
             else {
-                parts.push(linkTextPart(name + (suffix || text.indexOf("://") === 0 ? "" : " ") + text));

```

Comment:
before the `if`, `findLinkNameEnd` and `skipSeparatorFromLinkText` have basically the same intent: skip some intermediate text. Then this PR adds more skip logic, but only in the `else` case.

I think it should all be moved into a single place, preferably before the `if`. The combination of findLinkNameEnd/skipSeparateFromLinkText probably needs to track a start/end of the intermediate text instead of just the start.